### PR TITLE
[MLU-FIX] fix(nonzero): add sync and fix its test

### PR DESCRIPTION
**Pull request by claire199705**

### CODE REVIEW

File: backends/mlu/kernels/nonzero_kernel.cc [hunk 1]
```diff
@@ -38,6 +38,7 @@ void NonZeroKernel(const Context& dev_ctx,
                    GetBasePtr(&num_true));
 
   Tensor local_true_num;
+  dev_ctx.Wait();
```
Comment (ShawnNew) [inline backends/mlu/kernels/nonzero_kernel.cc]: add a note here for future reference

**Review by ShawnNew — CHANGES_REQUESTED** [review]

File: backends/mlu/kernels/nonzero_kernel.cc (hunk 1 above)
Comment (claire199705) [reply backends/mlu/kernels/nonzero_kernel.cc]: 已添加

Comment (claire199705) [reply backends/mlu/kernels/nonzero_kernel.cc]: already done

**Comment by ShawnNew** [comment]
👍

**Review by ShawnNew — APPROVED** [review]
👍

**Pull request merged**